app/controllers/company_satisfactions_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CompanySatisfactionsController < PagesController
+  def new
+    need = Need.find(params[:besoin])
+    satisfaction = CompanySatisfaction.find_by(need: need)
+    if satisfaction.present?
+      flash.notice = t('.company_satisfaction_exist')
+      redirect_to root_path
+    elsif need.nil? || need.diagnosis.visitee.email != params[:email]
+      redirect_to root_path
+    end
+    @company_satisfaction = CompanySatisfaction.new(need: need)
+  end
+
+  def create
+    @company_satisfaction = CompanySatisfaction.new(satisfaction_params)
+    if @company_satisfaction.save!
+      redirect_to thank_you_company_satisfactions_path
+    else
+      render :new
```

Comment:
A priori, pas trop de raisons que ça échoue, mais tu peux quand même afficher l’erreur le cas échéant, genre:
```suggestion
      redirect_to action: :new, alert: @company_satisfaction.errors.full_messages.to_sentence
```

Reply:
Euh, d’ailleurs, avec `save!` au dessus, on va avoir une exception en cas d’échec, pas un retour normal.